### ISSUE TITLE
OSRM-routed exits unexpectedly on some requests (MLD, with alternatives)

### ISSUE THREAD

**Issue by vbolshutkin** [Project-OSRM/osrm-backend#4723]
I am new to OSRM, was tried to launch it using Docker and followed the instructions in README.md. But when I started using `osrm-frontend` I noticed `osrm-routed` crashes (exits unexpectedly without any message, even with `-l DEBUG`) after some time of dragging marker over the map.

Further investigations have shown, that it reproduces consistently on concrete requests. For example, here are steps to reproduce
```
wget http://download.geofabrik.de/europe/germany/berlin-latest.osm.pbf
docker run -t -v $(pwd):/data osrm/osrm-backend osrm-extract -p /opt/car.lua /data/berlin-latest.osm.pbf
docker run -t -v $(pwd):/data osrm/osrm-backend osrm-partition /data/berlin-latest.osrm
docker run -t -v $(pwd):/data osrm/osrm-backend osrm-customize /data/berlin-latest.osrm
docker run -t -i -p 5000:5000 -v $(pwd):/data osrm/osrm-backend osrm-routed --algorithm mld /data/berlin-latest.osrm

curl "http://localhost:5000/route/v1/driving/13.4385964,52.5198535;13.410916328430176,52.50373584990983?overview=false&alternatives=true&steps=true&hints=QwICgP___381AAAAQQAAAHsAAAAvAAAANQAAAEEAAAB7AAAALwAAALwAAADIDs0A1mMhA4QOzQCuYyEDAwDPE09TUk0=;"
```

That URL was originally generated by `osrm-frontend`. At the same time, in most cases urls generated by the frontend simply work.

Also, if I simply remove `alternatives=true` for the problematic URL it seems to work.

I could only reproduce the issue with MLD algorithm.

**Comment by oxidase** [comment]
@vbolshutkin please could you try the query without `hints` parameter
```
curl "http://localhost:5000/route/v1/driving/13.4385964,52.5198535;13.410916328430176,52.50373584990983?overview=false&alternatives=true&steps=true"
```
and the latest docker image with v5.15.0 
```
osrm/osrm-backend   latest              sha256:78941e3b8b58035c0c5dd2338910a142158694d5e5d9eaceca5d98d6f0c318d4   01c2cd22c312        17 hours ago        20MB
```

There was a recent bug fix 6fd0b56e32bc0ec932d64dfb940022bbe4d85b5f in 5.14.1 release and can be related to the issue

**Comment by vbolshutkin** [comment]
@oxidase Thank you for quick reply, and you are right! My version was 5.14.0 and `docker pull osrm/osrm-backend:latest` helped. Now it works with both `alternatives` and `hints` parameters.
